Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -5177,6 +5182,22 @@ void webkit_web_view_get_snapshot(WebKitWebView* webView, WebKitSnapshotRegion r
                     g_task_return_pointer(task.get(), surface.leakRef(), reinterpret_cast<GDestroyNotify>(cairo_surface_destroy));
                     return;
                 }
+#endif
+#else
+                RELEASE_ASSERT(bitmap->bytesPerRow() <= std::numeric_limits<guint>::max());
+                const auto imageSize = bitmap->size();
+                const guint stride = bitmap->bytesPerRow();
+                auto imageBytes = [&]() -> GRefPtr<GBytes> {
+                    auto* b = bitmap.leakRef();
+                    return adoptGRef(g_bytes_new_with_free_func(b->span().data(), b->span().size(),
+                        [](void* data) {
+                            reinterpret_cast<ShareableBitmap*>(data)->deref();
+                        },
+                        b));
+                };
+                auto* image = webkitImageNew(imageSize.width(), imageSize.height(), stride, imageBytes());
```

Comment:
And here you could still use the bitmap instead of saving the size and stride and then WTFMove(imageBytes) instead.